ground-ingest/src/main/java/edu/berkeley/ground/ingest/GroundWriter.java

```diff
@@ -0,0 +1,109 @@
+package edu.berkeley.ground.ingest;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.commons.httpclient.HttpClient;
+import org.apache.commons.httpclient.methods.PostMethod;
+import org.codehaus.jackson.JsonGenerationException;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.map.JsonMappingException;
+import org.codehaus.jackson.map.ObjectMapper;
+import edu.berkeley.ground.api.models.Node;
+import edu.berkeley.ground.api.models.NodeVersion;
+import edu.berkeley.ground.api.models.Tag;
+import edu.berkeley.ground.api.versions.GroundType;
+import gobblin.writer.DataWriter;
+import java.net.URLEncoder;
+
+
+public class GroundWriter<D> implements DataWriter<GenericRecord>  {
+  
+    public void write(GenericRecord record) throws IOException {
+    
+      //extracting the metadata from the GenericRecord and storing it in the field tags      
+      Map<String, Tag> tags =  new HashMap<String, Tag>();
+      GroundType typeString = GroundType.STRING;
+      GroundType typeInt = GroundType.INTEGER;
+      Tag timeCreated = new Tag("id5", "timeCreated", record.get("timeCreated"), typeInt);
+      Tag length = new Tag("id6", "fileLength", record.get("length"), typeInt);
+      Tag modificationTime = new Tag("id7", "modificationTime", record.get("modificationTime"), typeInt);
+      Tag owner = new Tag("id8", "owner", record.get("owner"), typeString);
+      tags.put("timeCreated", timeCreated);
+      tags.put("length", length);
+      tags.put("modificationTime", modificationTime);
+      tags.put("owner", owner);
+        
+      //extracting the name of the file to create the Node
+      String name = record.get("name").toString();
+      node(name, tags);
+   
+  } 
+    
+    //method to create a Node given the name
+    public void node(String name, Map<String, Tag> tags) throws JsonGenerationException, JsonMappingException, IOException {
+     
+      Node node = new Node("id", name);
+      ObjectMapper mapper = new ObjectMapper();
+      String jsonString = mapper.writeValueAsString(node);
+      HttpClient client = new HttpClient();
```

Comment:
HttpClient should be member variable inside the writer, so you're not creating a new client per node.